Cabal/Distribution/Utils/ShortText/Internal.hs

```diff
@@ -0,0 +1,107 @@
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE DeriveDataTypeable #-}
+{-# LANGUAGE DeriveGeneric #-}
+
+module Distribution.Utils.ShortText.Internal
```

Comment:
Let me see if I understand correctly, because `Distribution.Compat.Prelude.Internal` is setup slightly differently than the situation here.

In the Prelude case, we have `Distribution.Compat.Prelude` which is NOT exposed, and then we have `Distribution.Compat.Prelude.Internal` which is exposed, but no one in Cabal library actually uses.

The situation is a little different for `Distribution.Utils.ShortText.Internal`, since it contains the real definitions for the types in question, which then are reexported (with some things hidden) by `Distribution.Utils.ShortText`. So, is the suggestion to have `Distribution.Utils.ShortText` suppress warnings deprecations, or is the suggestion to have a NOT exposed `Distribution.Utils.ShortText.ReallyInternal` with no warnings, `Distribution.Utils.ShortText.Internal` with warnings but not used by anyone, and have `Distribution.Utils.ShortText` import the not exposed things?